components/bookshelf/detailedbook/DetailedBook.tsx

```diff
@@ -0,0 +1,157 @@
+import { BookType } from "api/types"
+import {
+  BookContainer,
+  BookHeader,
+  BookImageContainer,
+  BookTitle,
+  BookTitleBox,
+  DateContainer,
+  Divider,
+  ItalicizedContainer,
+  ItalicizedText,
+  SideBtn,
+  SideButtonContainer,
+  Synopsis,
+  WhiteSpace,
+} from "./style"
+import React, { useState } from "react"
+import Image from "next/image"
+import { Rating } from "@mui/material"
+import Moment from "moment"
+import { setRating } from "api/books/setRating"
+import { ExtendedSideBar, SideButton } from "components/styled/bookshelf"
+import { markAsRead } from "api/books/markAsRead"
+import { removeFromList } from "api/books/removeFromList"
+import { setToRead } from "api/books/setToRead"
+import { addToReadingList } from "api/books/addToReadingList"
+import { checkRating, checkID, checkFinishDate } from "./utils"
+
+interface DetailedBookProps {
+  book: BookType
+}
+
+const DetailedBook = ({ book }: DetailedBookProps) => {
+  const [currentRating, setCurrentRating] = useState<number>(() =>
+    checkRating(book)
+  )
+  const [userBookID, setUserBookID] = useState<number | null>(() =>
+    checkID(book)
+  )
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [isFinished, setIsFinished] = useState<boolean>(() => {
+    return checkFinishDate(book)
+  })
+
+  const rateBook = (_event, newRating: number | null) => {
```

Comment:
Saw this function twice. You might want to have this as a shared function rather. Can you try putting it in `useBook`?

Same for the other duplicated code.